### PR TITLE
feat: Newer operator sdk

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### What does this PR do?
Use newer operator-sdk for che-operator.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/19466

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [x] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: bundle/nightly/eclipse-che-preview-openshift/bundle.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM scratch
+
+# Core bundle labels.
+LABEL operators.operatorframework.io.bundle.mediatype.v1=registry+v1
+LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
+LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
+LABEL operators.operatorframework.io.bundle.package.v1=eclipse-che-preview-openshift
+LABEL operators.operatorframework.io.bundle.channels.v1=nightly
+LABEL operators.operatorframework.io.bundle.channel.default.v1=nightly
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.6.1+git
```
Comment (tolusha) [inline bundle/nightly/eclipse-che-preview-openshift/bundle.Dockerfile:10]: May be 1.7?

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -100,17 +103,16 @@ spec:
           livenessProbe:
             httpGet:
               path: /healthz
-              port: 6789
+              port: 8081
```
Comment (tolusha) [inline config/manager/manager.yaml:106]: Can we use 6789 port ?

File: config/crd/bases/org_v1_che_crd-v1beta1.yaml [hunk 3]
```diff
@@ -19,6 +23,7 @@ spec:
     listKind: CheClusterList
     plural: checlusters
     singular: checluster
+  preserveUnknownFields: false
```
Comment (tolusha) [inline config/crd/bases/org_v1_che_crd-v1beta1.yaml:26]: Must be removed for crd-v1beta1

File: config/manager/manager.yaml [hunk 4]
```diff
@@ -23,19 +26,19 @@ spec:
     metadata:
       labels:
         app: che-operator
-        app.kubernetes.io/name: che
-        app.kubernetes.io/instance: che
         app.kubernetes.io/component: che-operator
+        app.kubernetes.io/instance: che
+        app.kubernetes.io/managed-by: olm
```
Comment (tolusha) [inline config/manager/manager.yaml:31]: Pls remove

File: config/manager/manager.yaml [hunk 5]
```diff
@@ -23,19 +26,19 @@ spec:
...
-          ports:
-            - containerPort: 60000
-              name: metrics
-          command:
-            - /usr/local/bin/che-operator
+        - command:
+            - /manager
+          args:
+            - --leader-elect
+          image: quay.io/aandriienko/che-operator:nightly
```
Comment (tolusha) [inline config/manager/manager.yaml:39]: Don't forget to revert.

File: pkg/deploy/identity-provider/deployment_keycloak_test.go [hunk 6]
```diff
@@ -301,9 +313,4 @@ func TestSyncKeycloakDeploymentToCluster(t *testing.T) {
 	if cmRevision == nil || cmRevision.Value != "1" {
 		t.Fatalf("Failed to sync deployment")
 	}
-
-	// check self-signed-certificate secret revision
-	if actual.ObjectMeta.Annotations["che.self-signed-certificate.version"] != "2" {
```
[the remark this answers is not in the document]
Comment (AndrienkoAleksandr) [reply pkg/deploy/identity-provider/deployment_keycloak_test.go:306]: I turn back and fixed test.

File: config/manager/manager.yaml (hunk 2 above)
Comment (AndrienkoAleksandr) [reply config/manager/manager.yaml]: Yes. Done

**Comment by nickboldt** [comment]
lol, >2300 files in a single PR? glad I don't have to review this one :)

**Review by tolusha — APPROVED** [review]

**Comment by tolusha** [comment]
Great work!

**Comment by tolusha** [comment]
`Minishift / Test Eclipse Che update from stable to nightly` passed locally.
We are good to merge here.

**Pull request merged**